cmd/site-replication.go

```diff
@@ -472,11 +541,17 @@ func (c *SiteReplicationSys) InternalJoinReq(ctx context.Context, arg madmin.SRI
 	if ourName == "" {
 		return errSRInvalidRequest(errSRSelfNotFound)
 	}
-
-	svcCred, err := globalIAMSys.NewServiceAccount(ctx, arg.SvcAcctParent, nil, newServiceAccountOpts{
-		accessKey: arg.SvcAcctAccessKey,
-		secretKey: arg.SvcAcctSecretKey,
-	})
+	var (
+		svcCred auth.Credentials
+		err     error
+	)
+	svcCred, _, err = globalIAMSys.GetServiceAccount(ctx, arg.SvcAcctAccessKey)
+	if err == errNoSuchServiceAccount {
+		svcCred, err = globalIAMSys.NewServiceAccount(ctx, arg.SvcAcctParent, nil, newServiceAccountOpts{
+			accessKey: arg.SvcAcctAccessKey,
+			secretKey: arg.SvcAcctSecretKey,
+		})
+	}
```

Comment:
is this kind of code needed? why not just add `blindly` ?

Reply:
if site replicator service account has already been created when setting up site replication earlier, we don't want to re-create it. Otherwise the remote targets set up on buckets would need to be updated whenever a new site is added because service account creds would have changed.